tests/async/test_async_client.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+import cohere
+
+
+@pytest.mark.asyncio
+async def test_async_enter():
+    cli = cohere.AsyncClient() 
+    async with cli as co:
+        await co.generate(model='medium', prompt='co:here', max_tokens=1)
+    assert cli._backend._session is None
+
+@pytest.mark.asyncio
+async def test_async_enter_invalid_key():
+    cli = cohere.AsyncClient(api_key="invalid") 
+    with pytest.raises(cohere.CohereError) as exc:
+        async with cli as co:
+            pass
+    assert "invalid API key" in str(exc)
+
+
```

Comment:
I have not run the formatter, since changing to black will make the diff even more of a nightmare than it is. don't worry about formattting, it will run after you're done.